handlers/validator.go

```diff
@@ -2006,32 +2015,71 @@ func ValidatorSync(w http.ResponseWriter, r *http.Request) {
 				startEpoch = lastShownEpoch - 1
 				endEpoch = lastShownEpoch + nextPeriodLimit - 1
 			}
-			nextPeriodSyncDuties, err := db.BigtableClient.GetValidatorSyncDutiesHistoryOrdered(validatorIndex, startEpoch, endEpoch, ascOrdering)
+			nextPeriodSyncDuties, err := db.BigtableClient.GetValidatorSyncDutiesHistoryOrdered([]uint64{}, startEpoch, endEpoch, ascOrdering)
 			if err != nil {
 				logger.Errorf("error retrieving second validator sync duty data from bigtable: %v", err)
 				http.Error(w, "Internal server error", http.StatusInternalServerError)
 				return
 			}
-			syncDuties = append(syncDuties, nextPeriodSyncDuties...)
+
+			nextPeriodSyncDutiesValidator := nextPeriodSyncDuties[validatorIndex]
+			nextPeriodSyncDutiesAllValidators := make([]uint64, nextPeriodLimit*utils.Config.Chain.Config.SlotsPerEpoch)
+			for _, duties := range nextPeriodSyncDuties {
+				for idx := range duties {
+					if duties[idx].Status == 1 {
+						nextPeriodSyncDutiesAllValidators[idx]++
+					}
+				}
+			}
```

Comment:
Instead of doing this loop twice (here and line 2000), can't you just do it once after the append?

Reply:
I can't append **syncDuties** and **nextPeriodSyncDuties** together because they are maps.
I also can't combine them to one map because the **validatorindex** is the key and there can be validators that were part of the sync committee in both periods.
The validator whose page we currently have open is guaranteed to be in both.